modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java

```diff
@@ -190,13 +200,27 @@ public FriendlyURLEntry deleteFriendlyURLEntry(
 			}
 			else {
 				friendlyURLEntryMapping.setFriendlyURLEntryId(
-					friendlyURLEntry.getFriendlyURLEntryId());
+					friendlyURLEntryId);
 
 				_friendlyURLEntryMappingPersistence.update(
 					friendlyURLEntryMapping);
 			}
 		}
 
+		// Asset
+
+		try {
+			AssetEntry assetEntry = _assetEntryLocalService.fetchEntry(
+				FriendlyURLEntry.class.getName(), friendlyURLEntryId);
+
+			if (assetEntry != null) {
+				_assetEntryLocalService.deleteEntry(assetEntry.getEntryId());
+			}
+		}
+		catch (PortalException portalException) {
+			throw new RuntimeException(portalException);
+		}
+
```

Comment:
I think we should bubble up the exception, without wrapping it. To avoid the breaking change, we could use `ReflectionUtil.throwException` (portal exn is caught upwards, so it is ok to use it here). But honestly, I think even the breaking change is fine. Pick your poison :-)

Also, can we add a private `_deleteAssetEntry(long)` method to hide the cruft? This fragment of code is duplicated multiple times in this same class.

Reply:
Done :)